delly: init at 0.8.1

scalavision opened the pull request:
<!-- Nixpkgs has a lot of new incoming Pull Requests, but not enough people to review this constant stream. Even if you aren't a committer, we would appreciate reviews of other PRs, especially simple ones like package updates. Just testing the relevant package/service and leaving a comment saying what you tested, how you tested it and whether it worked would be great. List of open PRs: <https://github.com/NixOS/nixpkgs/pulls>, for more about reviewing contributions: <https://hydra.nixos.org/job/nixpkgs/trunk/manual/latest/download/1/nixpkgs/manual.html#sec-reviewing-contributions>. Reviewing isn't mandatory, but it would help out a lot and reduce the average time-to-merge for all of us. Thanks a lot if you do! -->
###### Motivation for this change


###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [ x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ x] Tested execution of all binary files (usually in `./result/bin/`)
- [ x] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ x] Ensured that relevant documentation is up to date
- [ x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @

scalavision committed:
delly: init at 0.8.1

scalavision commented:
Delly is a very much used tool used to find structural variants within dna sequenced, short read data. It would be very nice to have it in nixpkgs.

markuskowa commented on pkgs/applications/science/biology/delly/default.nix:
```diff
@@ -0,0 +1,44 @@
+  pname = "delly";
+  version = "0.8.1";
+
+  src = fetchgit {
```
Please use `fetchFromGitHub` here.

markuskowa commented on pkgs/applications/science/biology/delly/default.nix:
```diff
@@ -0,0 +1,44 @@
+
+  buildInputs = [ zlib htslib bzip2 lzma ncurses boost ];
+
+  buildPhase = ''
```
`preBuild` is the better choice here:
```suggestion
  preBuild = ''
     export EBROOTHTSLIB=${htslib}
  '';   
```

veprbl replied:
No need to use export, just set the attribute:
```nix
stdenv.mkDerivation {
  # ...

  EBROOTHTSLIB = htslib;

  # ...
}
```

markuskowa commented on pkgs/applications/science/biology/delly/default.nix:
```diff
@@ -0,0 +1,44 @@
+  '';
+
+  meta = with stdenv.lib; {
+    description = "Structural variant caller";
```
Maybe something like that?
```suggestion
    description = "Structural variant caller for DNA sequence matching";
```
That would give it more context if viewed in a list of packages.

markuskowa commented on pkgs/applications/science/biology/delly/default.nix:
```diff
@@ -0,0 +1,44 @@
+  installPhase = ''
+      mkdir -p $out/bin
+      cp src/delly $out/bin/delly    
+      export PATH=$PATH:$out/bin/delly
```
What is the purpose of the `export`?

veprbl commented on pkgs/applications/science/biology/delly/default.nix:
```diff
@@ -0,0 +1,44 @@
+
+  installPhase = ''
+      mkdir -p $out/bin
+      cp src/delly $out/bin/delly    
```
Can replace these two lines
```suggestion
      install -Dm555 src/delly $out/bin/delly    
```

veprbl commented on pkgs/applications/science/biology/delly/default.nix:
```diff
@@ -0,0 +1,44 @@
+      make
+  '';
+
+  installPhase = ''
```
When overriding a stdenv phase it's a good practice to call the standard hooks:
```nix
stdenv.mkDerivation {
  # ...

  installPhase = ''
    runHook preInstall

    # your installation procedure here

    runHook postInstall
  '';

  # ...
}
```

veprbl commented on pkgs/applications/science/biology/delly/default.nix:
```diff
@@ -0,0 +1,44 @@
+
+stdenv.mkDerivation rec {
+
+  name = "${pname}-${version}";
```
Remove this line, this is already done inside the `mkDerivation`.

scalavision replied:
I removed the `name` attribute, is this the reason why the build failed?

mmahut commented:
After fixing the issues above, please squash your commits too.

scalavision commented:
Thanks for all the nice feedback! I've removed buildPhase, only using prePhase. Also I've cleaned up the installPhase, which seems to work.

Anyone knows why the checkMeta validation doesn't work? It says something about scylla-db. I don't think I have anything related to that.

veprbl committed:
fixup

prePhase is not a thing

veprbl commented:
Eval should be fixed by c0d5833156cd32086642207fbb026ab772d16822

veprbl commented:
@GrahamcOfBorg build delly
